nav2_dwb_controller/dwb_core/src/publisher.cpp

```diff
@@ -235,8 +247,12 @@ DWBPublisher::publishLocalPlan(
     nav_2d_utils::poses2DToPath(
       traj.poses, header.frame_id,
       header.stamp));
-  if (node_->count_subscribers(local_pub_->get_topic_name()) > 0) {
-    local_pub_->publish(std::move(path));
+
+  {
+    auto node = node_.lock();
+    if (node->count_subscribers(local_pub_->get_topic_name()) > 0) {
```

Comment:
http://docs.ros2.org/dashing/api/rclcpp/classrclcpp_1_1PublisherBase.html the publisher can get the number of subscribers itself. I think that would avoid needing to store the node in a few classes or locking to do operations like this.